cmd/cmd.go

```diff
@@ -280,12 +284,128 @@ func createBlob(cmd *cobra.Command, client *api.Client, path string) (string, er
 	}
 
 	digest := fmt.Sprintf("sha256:%x", hash.Sum(nil))
+
+	// Here, we want to check if the server is local
+	// If true, call, createBlobLocal
+	// This should find the model directory, copy blob over, and return the digest
+	// If this fails, just upload it
+	// If this is successful, return the digest
+
+	// Resolve server to IP
+	// Check if server is local
+	/* if client.IsLocal() {
+		digest = strings.ReplaceAll(digest, ":", "-")
+		config, err := client.HeadBlob(cmd.Context(), digest)
+		if err != nil {
+			return "", err
+		}
+
+		modelDir := config.ModelDir
+
+		// Get blob destination
+
+		dest := filepath.Join(modelDir, "blobs", digest)
+
+		err = createBlobLocal(path, dest)
+		if err == nil {
+			return digest, nil
+		}
+	} */
+	if client.IsLocal() {
+		dest, err := getLocalPath(cmd.Context(), digest)
+
+		if errors.Is(err, ErrBlobExists) {
+			return digest, nil
+		}
+
+		if err == nil {
+			err = createBlobLocal(path, dest)
```

Comment:
What happens if this fails? Wouldn't we still want to upload the file via the http server?

Reply:
If this fails, it continues onto the default "client.CreateBlob" call

Reply:
The only time it exits this function call is if a) it finds the blob already during "getLocalPath" or b) successfully copies the blob over locally